src/EventStore.Projections.Core/Services/v8/V8ProjectionStateHandler.cs

```diff
@@ -126,17 +129,24 @@ public class EmittedEventJsonContract {
 			CheckpointTag eventPosition, string category, ResolvedEvent @event) {
 			CheckDisposed();
 			if (@event == null) throw new ArgumentNullException("event");
-			var partition = _query.GetPartition(
-				GetEventData(@event),
-				new string[] {
-					@event.EventStreamId, @event.IsJson ? "1" : "", @event.EventType, category ?? "",
-					@event.EventSequenceNumber.ToString(CultureInfo.InvariantCulture), @event.Metadata ?? "",
-					@event.PositionMetadata ?? ""
-				});
+
+			string eventData = GetEventData(@event);
+
+			if (string.IsNullOrEmpty(eventData)) {
```

Comment:
Thanks for the PR @StevenBlair123!

I've tested it locally and it does fix the AccessViolation issue, however there are cases where events with empty data should still be handled.
For example, handling the metadata of an empty non-json event.

It would be better to check the EventType of the ResolvedEvent, which is an empty string for unresolved LinkTos

For example:

```suggestion
			if (string.IsNullOrEmpty(@event.EventType)) {
```

Reply:
Made the changed and push.